libs/ledger-live-common/src/families/icon/js-synchronisation.ts

```diff
@@ -0,0 +1,50 @@
+import { encodeAccountId } from "../../account";
+import type { GetAccountShape } from "../../bridge/jsHelpers";
+import { makeSync, makeScanAccounts, mergeOps } from "../../bridge/jsHelpers";
+
+import { getAccount, getOperations } from "./api";
+import BigNumber from "bignumber.js";
+import { getApiUrl } from "./logic";
+
+const getAccountShape: GetAccountShape = async (info) => {
+  const { address, initialAccount, currency, derivationMode } = info;
+  const accountId = encodeAccountId({
+    type: "js",
+    version: "2",
+    currencyId: currency.id,
+    xpubOrAddress: address,
+    derivationMode,
+  });
+  const oldOperations = initialAccount?.operations || [];
+  // Needed for incremental synchronisation
+  const startAt = 1;
```

Comment:
_Incremental synchronization_ means fetching only the new operations that aren't already known in the account, as an optimization.

To implement it, `startAt` should be something that you can use as an offset with the API that's fetching the account's transactions. Most of the time it can be the block height of the first operation in `oldOperations`, however you should check what is supported with the API and see if `startAt` can take a relevant value (currently it is useless).